app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.kt

```diff
@@ -0,0 +1,167 @@
+     * @param dialog the dialog to be shown
+     */
+    private fun showSafely(activity: Activity?, dialog: Dialog?) {
+
+        if (activity == null || dialog == null) {
+            Timber.d("Show called with null activity / dialog. Ignoring.")
+            return
+        }
+
+        if (activity.isFinishing || activity.isDestroyed) {
+            Timber.e("Activity is not running. Could not show dialog. ")
+            return
+        }
+        try {
+            dialog.show()
+        } catch (e: IllegalStateException) {
+            Timber.e(e, "Could not show dialog.")
+        }
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?
+    ) {
+        showAlertDialog(
+            activity,
+            title,
+            message,
+            activity.getString(R.string.yes),
+            activity.getString(R.string.no),
+            onPositiveBtnClick,
+            onNegativeBtnClick
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String,
+        negativeButtonText: String,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?
+    ) {
+        showAlertDialog(
+            activity,
+            title,
+            message,
+            positiveButtonText,
+            negativeButtonText,
+            onPositiveBtnClick,
+            onNegativeBtnClick
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?,
+        customView: View,
+        cancelable: Boolean
+    ) {
+        showAlertDialog(
+            activity,
+            title,
+            message,
+            activity.getString(R.string.yes),
+            activity.getString(R.string.no),
+            onPositiveBtnClick,
+            onNegativeBtnClick,
+            customView,
+            cancelable
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String?,
+        onPositiveBtnClick: Runnable?,
+        cancellable: Boolean
+    ) {
+        showAlertDialog(
+            activity,
+            title,
+            message,
+            positiveButtonText,
+            onPositiveBtnClick,
+            cancellable
+        )
+    }
+
+    /**
+     * show a dialog
+     * @param activity
+     * @param title
+     * @param message
+     * @param positiveButtonText
+     * @param negativeButtonText
+     * @param onPositiveBtnClick
+     * @param onNegativeBtnClick
+     * @param customView
+     * @param cancelable
+     */
+    @JvmStatic
+    fun showAlertDialog(
```

Comment:
This is quite broken. Only 1 of these methods actually calls this, the others just call themselves in an infinite loop and then would overflow the stack probably

Reply:
My bad, I thought if I skipped the parms then the default value will automatically be picked and the function will be called but now I see my mistake.

Reply:
@macgills I have made the changes. This is the first conversion PR and I am learning kotlin so wasn't smooth but next PRs will be smooth 😅

Reply:
It is a learning exercise, don't worry

Reply:
We did want to take advantage of the default arguments so your instincts were correct. It is because all these methods have the same name that this didn't work out. We could fix this by simply renaming the internal private method to `createAndShowDialogSafely`.

The API of this function is still unfortunately quite difficult to work with but this is because everything is calling it from java after all. I think my current design on kiwix is good from the api perspective, if you have curiosity about what this can be like in kotlin then I would suggest a quick read of [AlertDialogShower](https://github.com/kiwix/kiwix-android/blob/develop/core/src/main/java/org/kiwix/kiwixmobile/core/utils/AlertDialogShower.kt), [KiwixDialog](https://github.com/kiwix/kiwix-android/blob/develop/core/src/main/java/org/kiwix/kiwixmobile/core/utils/KiwixDialog.kt) which allow the showing of a dialog to be as simple as
```
dialogShower.show(StopDownload, { downloader.cancelDownload(it.downloadId) })
```

To be clear this is not saying you have to rewrite this class to look like those classes, that would be out of scope for this ticket